emily/handler/tests/integration/deposit.rs

```diff
@@ -1628,13 +1651,23 @@ async fn emily_process_deposit_updates_when_some_of_them_are_unknown() {
     .expect("Received an error after making a valid update deposit request api call.");
 
     // Check that multistatus response is returned correctly.
-    assert!(update_responce.deposits.iter().all(|deposit| {
-        if deposit.deposit.bitcoin_txid == create_deposit_body1.bitcoin_txid {
-            deposit.status == 200
-        } else {
-            deposit.status == 404
-        }
-    }));
+    update_responce
+        .deposits
+        .iter()
+        .for_each(|deposit| match &deposit.deposit {
+            Some(Some(inner)) => {
+                assert_eq!(inner.bitcoin_txid, create_deposit_body1.bitcoin_txid);
+                assert_eq!(deposit.status, 200);
```

Comment:
nit: for completeness, can we also check that `error` is none? also in withdrawal